lib/pipekit/deal.rb

```diff
@@ -2,8 +2,9 @@ module Pipekit
   class Deal
     include Repository
 
-    def get_by_person_id(person_id)
-      request.get("/#{Pipekit::Person.uri}/#{person_id}/#{self.class.uri}")
+    def get_by_person_id(person_id, request = nil)
+      request ||= Request.new(Person.resource)
```

Comment:
yeah it's just for testing purposes, I guess I could stub a call to Request.new but DI in methods is pretty standard as far as I've seen. Can you think of an alternative?

Thinking about it often we end up doing DI pretty much just for testing purposes